kie-server-parent/kie-server-api/src/main/java/org/kie/server/api/marshalling/xstream/KieServerTypePermission.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2017 - 2017 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+package org.kie.server.api.marshalling.xstream;
+
+import static org.kie.server.api.KieServerConstants.SYSTEM_XSTREAM_ENABLED_PACKAGES;
+
+import java.util.Set;
+
+import org.kie.internal.utils.KieTypeResolver;
+import com.thoughtworks.xstream.security.WildcardTypePermission;
+
+/**
+ * Kie Server specific type permission implementation that allows:
+ * <ul>
+ *  <li>org.kie.server.api.model classes (including subpackages)</li>
+ *  <li>classes that come from kjar or its dependencies (were loaded by kjar class loader)</li>
+ *  <li>set of classes explicitly given when constructing this instance</li>
+ *  <li>optionally defined by wildcard that is given via system property: org.kie.server.xstream.enabled.packages (comma separated list of wildcard expressions)</li>
+ * </ul>
+ *
+ */
+public class KieServerTypePermission extends WildcardTypePermission {
+    
+    private static final String[] WHITELISTED_PACKAGES = new String[]{
+                                                                      "org.kie.server.api.model.**"
+    };
+
+    private Set<Class<?>> classes;
+    
+    public KieServerTypePermission(Set<Class<?>> classes) {
+        super(patterns());
+        this.classes = classes;
```

Comment:
@mswiderski if you want you could prevent a NPE by creating a new Set in case you're passed a null one. Also I think that it could be a good idea to declare the Set classes final.